packages/blocks/src/api/raw-handling/paste-handler.js

```diff
@@ -180,10 +183,16 @@ export function pasteHandler( {
 		HTML = HTML.normalize();
 	}
 
+	const isPlainText = ! HTML || isPlain( HTML );
+
+	if ( preservePlainText && isPlainText ) {
```

Comment:
Would there be any problem with removing the `isPlainText` check here, so that HTML is also returned as plain text in the code block?

Reply:
Pasted HTML should not be pasted as plaintext. You might for example paste some code from an HTML page, which would already be encoded and formatted correctly.